packages/integrations/vue/src/index.ts

```diff
@@ -80,6 +84,20 @@ export const setup = (app) => {
 				return `export const setup = () => {};`;
 			}
 		},
+		// Ensure that Vue components reference appEntrypoint directly
+		// This allows Astro to assosciate global styles imported in this file
+		// with the pages they should be injected to
+		transform(code, id) {
+			if (!appEntrypoint) return;
+			if (id.endsWith('.vue')) {
+				const s = new MagicString(code);
+				s.prepend(`import "${appEntrypoint}";\n`);
```

Comment:
If the change of resolving is related to here, I think it's safer to use `JSON.stringify` here instead like above so it's safer.

Reply:
Good catch!